### DISCUSSION TITLE
Reading UTF16 logs

### DISCUSSION THREAD

**Discussion by garethparris** [grafana-cold-storage/agent#4378] [Q&A] [answered]
How do I get Grafana agent component local.source.file to read and decode a file that is UTF16 ? With Promtail I could specify the encoding type so that when I query in Grafana my text searches work correctly.

**Comment by tpaschalis** [comment, marked as answer]
We've updated the `loki.source.file` with a new [Argument](https://grafana.com/docs/agent/next/flow/reference/components/loki.source.file/#arguments) to handle different encodings. This is already available in main and will be out with our next release later in August.

**Reply by garethparris** [reply]
Great thanks! I'll be able to get rid of my mssql server regex queries with that.
